ios/Classes/PdftronFlutterPlugin.m

```diff
@@ -1167,6 +1178,183 @@ - (void)getPageCropBox:(NSNumber *)pageNumber resultToken:(FlutterResult)result
+
+        tool.backToPanToolAfterUse = !self.continuousAnnotationEditing;
+
+        if ([tool isKindOfClass:[PTFreeHandCreate class]]
+            && ![tool isKindOfClass:[PTFreeHandHighlightCreate class]]) {
+            ((PTFreeHandCreate *)tool).multistrokeMode = self.continuousAnnotationEditing;
+        }
+    }
+
+    result(nil);
+}
+
+- (void)setFlagForFields:(NSArray <NSString *> *)fieldNames flag:(NSNumber *)flag flagValue:(bool)flagValue resultToken:(FlutterResult)result
+{
+    PTDocumentViewController *docVC = [self getDocumentViewController];
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        result([FlutterError errorWithCode:@"set_flag_for_fields" message:@"Failed to set flag for fields" details:@"Error: The document view controller has no document."]);
+        return;
+    }
+
+    PTPDFViewCtrl *pdfViewCtrl = docVC.pdfViewCtrl;
+    PTFieldFlag fieldFlag = (PTFieldFlag)flag.intValue;
+    NSError *error;
+
+    [pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        for (NSString *fieldName in fieldNames) {
+            PTField *field = [doc GetField:fieldName];
+            if ([field IsValid]) {
+                [field SetFlag:fieldFlag value:flagValue];
+                [pdfViewCtrl UpdateWithField:field];
+            }
+        }
+    } error:&error];
+
+    if (error) {
+        NSLog(@"Error: Failed to set field flags to doc. %@", error.localizedDescription);
+        result([FlutterError errorWithCode:@"set_flag_for_fields" message:@"Failed to set flag for fields" details:@"Error: Failed to set field flags to doc."]);
+    }
+
+    result(nil);
+}
+
+- (void)setValuesForFields:(NSString *)fieldWithValuesString resultToken:(FlutterResult)result
+{
+    PTDocumentViewController *docVC = [self getDocumentViewController];
+    NSArray *fieldWithValues = [PdftronFlutterPlugin PT_idAsArray:[PdftronFlutterPlugin PT_JSONStringToId:fieldWithValuesString]];
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        result([FlutterError errorWithCode:@"set_value_for_fields" message:@"Failed to set value for fields" details:@"Error: The document view controller has no document."]);
+        return;
+    }
+
+    PTPDFViewCtrl *pdfViewCtrl = docVC.pdfViewCtrl;
+    NSError *error;
+
+    [pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+
+        for (NSDictionary *fieldWithValue in fieldWithValues) {
+            NSString *fieldName = [PdftronFlutterPlugin PT_idAsNSString:fieldWithValue[PTFieldNameKey]];
+            id fieldValue = fieldWithValue[PTFieldValueKey];
+            PTField *field = [doc GetField:fieldName];
+
+            if ([field IsValid]) {
+                [self setFieldValue:field value:fieldValue pdfViewCtrl:pdfViewCtrl];
```

Comment:
Have you tested this? Do we not need a call to update here too?

Reply:
update is done in setFieldName, the helper function below it